lms/djangoapps/grades/course_grade_factory.py

```diff
@@ -169,13 +159,15 @@ def _update(user, course_data, force_update_subsections=False):
         Sends a COURSE_GRADE_CHANGED signal to listeners and a
         COURSE_GRADE_NOW_PASSED if learner has passed course.
         """
+        should_persist = should_persist_grades(course_data.course_key)
+
+        if should_persist and force_update_subsections:
+            prefetch(user, course_data.course_key)
```

Comment:
**Note to reviewers:** There might be a better place to do this, but for now, this top-level `_update` method prefetches all user-course data that helps reduce the number of queries when doing a bulk-update of all subsections.